CMake/sitkUseGTest.cmake

```diff
@@ -33,7 +33,9 @@ function(_sitk_gtest_use_gtest_source)
 
   set(BUILD_GTEST                 ON )
   set(BUILD_GMOCK                 OFF)
-
+  if ("${WIN32}")
+      set(gtest_disable_pthreads      ON CACHE INTERNAL "Disable use of pthreads in gtest")
+  endif()
```

Comment:
I tested with mingw. The problem does not appear to be with the `if(MINGW)` but with not setting the variable as `CACHE`.
https://github.com/SimpleITK/SimpleITK/blob/08a81298823ecc50ab952a4c63a01a8d692d4643/CMake/sitkUseGTest.cmake#L40

This seems familiar, I may have had these changes local and forgot to commit them.